### ISSUE TITLE
Lock adaptor versions by default

### ISSUE THREAD

**Issue by josephjclark** [OpenFn/lightning#2843]
## The Problem

Right now, by default, Lightning will save the adaptor version on each step as `@latest`. This is presented to users like this:

![Image](https://github.com/user-attachments/assets/cc0d08cb-89ee-4cca-9c35-d296733cb4c9)

The `@latest` flag is resolved at runtime - so new work orders from the same workflow, by default, may use different adaptor versions.

This is terrifying. I didn't realise we did this! Stu and I clearly miscommunicatid over this a long time ago.

The problem is that we release breaking adaptor versions, with major version bumps, all the time. We have to. But this puts workflows with `@latest` at risk because one day, maybe in 5 years time, they'll break overnight without warning.

Update: This caused production code to break in the latest Salesforce release. So it's a real problem.

##  The Solution

Here is what we're going to do:

* New steps, by default, should select the latest adaptor version - but not `@latest`
* `@latest`  should appear at the top of the picklist, but not selected by default.  
* We must ensure that when an step is selected in the diagram, the UI shows the _actual_ version number, and not just the `latest` one as a default (I think this is fine but we need to double check).
* We display a warning to users who select `@latest` to notify them that future breaking changes to the adaptor may cause their workflow to break.

Basically, by default, I want all steps to have version locked adaptors and not use a rolling `@latest` tag. But we are going to allow users to opt-in to `@latest` if they really want to (your funeral, pal).

This is much safer from a workflow stability point of view. But I will not that there's a bit of a security hit. When we release security patches, users will not automatically receive them.

## Other Stuff

* We should migrate all `@latest` in the database right now to a concrete version number?
* Related but different: we should consider a policy of auto-migration to the latest non-major version. I don't know how we'll deal with this yet - see #2844 for this and other thoughts

**Comment by josephjclark** [comment]
I worked on some customer code (MSF) today and I can see that adaptor version drift is going to hit them soon. When we release openMRS 5, their workflow will blow up.

Maybe in addition the steps above, we need to migrate the database to lock the version of anyone on currently on latest. Any users who really really want a floating version will just have to manually set the version to `@latest`. 

I still think #2844 needs to be tied into this.

This is super important.
